File: profiles/cclcc/hud/titlemenu.js
root.TitleMenu = {
    Type: TitleMenuType.CCLCC,
    DrawType: DrawComponentType.SystemMenu,
    PressToStartX: 823,
    PressToStartY: 749,
    PressToStartAnimDurationIn: 0.7,
    PressToStartAnimDurationOut: 0.7,
    PressToStartSprite: "TitleMenuPressToStart",
    BackgroundSprite: "TitleMenuBackground",
    MainBackgroundSprite: "MainMenuBackground",
    CopyrightTextSprite: "CopyrightText",
    CopyrightTextX: 566,
    CopyrightTextY: 955,
    MenuSprite: "TitleMenuMenu",
    MenuX: 27,
    MenuY: 26,
    ItemHighlightSprite: "TitleMenuItemHighlight",
    ItemHighlightOffsetX: 174,
    ItemHighlightOffsetY: 7,
    ItemHighlightPointerSprite: "TitleMenuPointerItemHighlight",
    ItemHighlightPointerY: 89,
    ItemPadding: 56,
    ItemYBase: 335,
    MenuEntriesSprites: [],
    MenuEntriesHighlightedSprites: [],
    MenuEntriesNum: 5
};

for (var i = 0; i < 5; i++) {
    root.Sprites["TitleMenuEntry" + i] = {
        Sheet: "TitleChip",
        Bounds: {
            X: 33,
            Y: 288 + i * root.TitleMenu.ItemPadding,
            Width: 223,
            Height: 37
        }
    };
    root.TitleMenu.MenuEntriesSprites.push("TitleMenuEntry" + i);
};

for (var i = 0; i < 5; i++) {
    root.Sprites["TitleMenuEntry" + i] = {
        Sheet: "TitleChip",
        Bounds: {
            X: 33,
            Y: 288 + i * root.TitleMenu.ItemPadding,
            Width: 223,
            Height: 37
        }
    };
    root.TitleMenu.MenuEntriesHighlightedSprites.push("TitleMenuEntry" + i);
};

root.Sprites["TitleMenuItemHighlight"] = {
    Sheet: "TitleChip",
    Bounds: { X: 896, Y: 64, Width: 638, Height: 50 },
};

root.Sprites["TitleMenuPointerItemHighlight"] = {
    Sheet: "TitleChip",
    Bounds: { X: 805, Y: 78, Width: 56, Height: 36 },
};

root.Sprites["TitleMenuBackground"] = {
    Sheet: "Title",
    Bounds: { X: 0, Y: 0, Width: 1920, Height: 1080 },
};

root.Sprites["MainMenuBackground"] = {
    Sheet: "Title",
    Bounds: { X: 0, Y: 1080, Width: 1920, Height: 1080 },
};

root.Sprites["TitleMenuPressToStart"] = {
    Sheet: "TitleChip",
    Bounds: { X: 583, Y: 75, Width: 198, Height: 169 },
};

root.Sprites["CopyrightText"] = {
    Sheet: "TitleChip",
    Bounds: { X: 582, Y: 9, Width: 785, Height: 21 },
};

root.Sprites["TitleMenuMenu"] = {
    Sheet: "TitleChip",
    Bounds: { X: 0, Y: 0, Width: 544, Height: 248 },
};

File: src/profile/games/cclcc/titlemenu.cpp
#include "titlemenu.h"
#include "../../../log.h"
#include "../../../window.h"
#include "../../profile_internal.h"

#include "../../ui/titlemenu.h"
#include "../../../game.h"
#include "../../../ui/ui.h"
#include "../../../games/cclcc/titlemenu.h"

namespace Impacto {
namespace Profile {
namespace CCLCC {
namespace TitleMenu {

Sprite BackgroundSprite;
Sprite MainBackgroundSprite;
Sprite CopyrightTextSprite;
Sprite MenuSprite;
Sprite ItemHighlightSprite;
Sprite ItemHighlightPointerSprite;

float CopyrightTextX;
float CopyrightTextY;
float MenuX;
float MenuY;
float ItemHighlightOffsetX;
float ItemHighlightOffsetY;
float ItemYBase;
float ItemPadding;
float ItemHighlightPointerY;

static void GetMemberSpriteArray(Sprite* arr, uint32_t count,
                                 char const* name) {
  EnsurePushMemberOfType(name, kArrayType);

  if (TopVal().Size() != count) {
    ImpLog(LL_Fatal, LC_Profile, "Expected to have %d sprites for %s\n", count,
           name);
    Window::Shutdown();
  }

  for (uint32_t i = 0; i < count; i++) {
    arr[i] = EnsureGetArrayElementSprite(i);
  }

  Pop();
}

void Configure() {
  BackgroundSprite = EnsureGetMemberSprite("BackgroundSprite");
  MainBackgroundSprite = EnsureGetMemberSprite("MainBackgroundSprite");
  CopyrightTextSprite = EnsureGetMemberSprite("CopyrightTextSprite");
  CopyrightTextX = EnsureGetMemberFloat("CopyrightTextX");
  CopyrightTextY = EnsureGetMemberFloat("CopyrightTextY");
  MenuSprite = EnsureGetMemberSprite("MenuSprite");
  MenuX = EnsureGetMemberFloat("MenuX");
  MenuY = EnsureGetMemberFloat("MenuY");
  ItemHighlightSprite = EnsureGetMemberSprite("ItemHighlightSprite");
  ItemHighlightOffsetX = EnsureGetMemberFloat("ItemHighlightOffsetX");
  ItemHighlightOffsetY = EnsureGetMemberFloat("ItemHighlightOffsetY");
  ItemYBase = EnsureGetMemberFloat("ItemYBase");
  ItemPadding = EnsureGetMemberFloat("ItemPadding");
  ItemHighlightPointerSprite =
      EnsureGetMemberSprite("ItemHighlightPointerSprite");
  ItemHighlightPointerY = EnsureGetMemberFloat("ItemHighlightPointerY");

  UI::CCLCC::TitleMenu* menu = new UI::CCLCC::TitleMenu();
  menu->PressToStartAnimation.DurationIn =
      Profile::TitleMenu::PressToStartAnimDurationIn;
  menu->PressToStartAnimation.DurationOut =
      Profile::TitleMenu::PressToStartAnimDurationOut;
  menu->PressToStartAnimation.LoopMode = ALM_ReverseDirection;

  UI::TitleMenuPtr = menu;

  auto drawType = Game::DrawComponentType::_from_integral_unchecked(
      EnsureGetMemberInt("DrawType"));

  UI::Menus[drawType].push_back(UI::TitleMenuPtr);
}

}  // namespace TitleMenu
}  // namespace CCLCC
}  // namespace Profile
}  // namespace Impacto